src/MergePlugin.php

```diff
@@ -430,6 +448,66 @@ public function onDependencySolve(InstallerEvent $event)
     }
 
     /**
+     * Handle an event callback following installation of a new package by
+     * checking to see if the package that was installed was our plugin.
+     *
+     * @param PackageEvent $event
+     */
+    public function onPostPackageInstall(PackageEvent $event)
+    {
+        $package = $event->getOperation()->getPackage()->getName();
+        if ($package === self::PACKAGE_NAME) {
+            $this->debug('composer-merge-plugin installed');
+            $this->pluginFirstInstall = true;
+        }
+    }
+
+    /**
+     * Is this the first time that the plugin has been installed?
+     *
+     * @return bool
+     */
+    public function isFirstInstall()
+    {
+        return $this->pluginFirstInstall;
+    }
+
+    /**
+     * Handle an event callback following an install or update command. If our
+     * plugin was installed during the run then trigger an update command to
+     * process any merge-patterns in the current config.
+     *
+     * @param Event $event
+     */
+    public function onPostInstallOrUpdate(Event $event)
+    {
+        if ($this->pluginFirstInstall) {
+            $this->pluginFirstInstall = false;
+            $this->debug(
+                '<comment>' .
+                'Running additional update to apply merge settings' .
+                '</comment>'
+            );
+
+            $installer = Installer::create(
+                $event->getIO(),
+                // Create a new Composer instance to ensure full processing of
+                // the merged files.
+                Factory::create($event->getIO(), null, false)
+            );
+
+            // Force update mode so that new packages are processed rather
+            // than just telling the user that composer.json and composer.lock
+            // don't match.
+            $installer->setUpdate(true);
+            $installer->setDevMode($event->isDevMode());
+            // TODO: can we set more flags to match the current run?
```

Comment:
I don't think that's possible right now, unless maybe by creating a new Console Input instance and checking for input options that were passed to the installer. Ideally we should extract most of the code in InstallCommand/UpdateCommand into standalone classes that could be reused here and tested too..